includes/class-sensei-course-enrolment-manager.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * File containing the class Sensei_Course_Enrolment_Manager.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles the management of course enrolment.
+ */
+class Sensei_Course_Enrolment_Manager {
+	const COURSE_ENROLMENT_SITE_SALT_OPTION = 'sensei_course_enrolment_site_salt';
+
+	/**
+	 * All course enrolment providers.
+	 *
+	 * @var Sensei_Course_Enrolment_Provider_Interface[]
+	 */
+	private static $enrolment_providers;
+
+	/**
+	 * Gets the descriptive name of the provider by ID.
+	 *
+	 * @param string $provider_id Unique identifier of the enrolment provider.
+	 *
+	 * @return string|false
+	 */
+	public static function get_enrolment_provider_name_by_id( $provider_id ) {
```

Comment:
Why do we need to define the methods as static? The only reason that I can think of is that we can access all this functionality easily from everywhere without getting hold of an instance.

If this is the only reason I think that its worth it to turn these to simple methods as it will allow us to initialize the internal state of the class properly.

Reply:
Talked about this in the earlier comment.